src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -245,9 +247,15 @@ private Delegate GetGetterCore(DataViewRow input, int iinfo, out Action disposer
             disposer = null;
             input.Schema.TryGetColumnIndex(_columns[iinfo].InputColumnName, out int srcCol);
             var srcType = input.Schema[srcCol].Type;
-            if (!(srcType is VectorDataViewType vectorType))
-                return ComposeGetterOne(input, iinfo, srcCol, srcType);
-            return ComposeGetterVec(input, iinfo, srcCol, vectorType);
+            if (GetVersionInfo().VerWrittenCur == 0x00010002)
```

Comment:
>GetVersionInfo [](start = 16, length = 14)

This will always return `0x00010003`.

Reply:
I think what you need is to add a field to `HashingTransformer` that has the version. That field will be initialized to be `0x00010003` in the constructors in lines 166 and 177, and according to the `ModelLoadContext`'s version info in the constructor in line 267.

---
In reply to: [387066236](https://github.com/dotnet/machinelearning/pull/4596#discussion_r387066236) [](ancestors = 387066236)

Reply:
field GetVersionInfo().VerWrittenCur will always return 0x00010003 for newly created models, this was meant for already existed models and those models will not be able to be converted to ONNX

Reply:
`GetVersionInfo()` calls the method in line 121, so it will return `0x00010003` even if the transformer was created from a model with a different version. When the `GetGetterCore` method is called you no longer have access to the model that created the transformer (or, it might not have been created from a serialized model at all), which is why you need to add a field to store the version, and initialize it in all the constructors. In the 

---
In reply to: [396068943](https://github.com/dotnet/machinelearning/pull/4596#discussion_r396068943) [](ancestors = 396068943)